README
b-datum-backup-agent-linux-perl
============

Simples agente de backup escrito em Perl para realizar backups na b-datum (http://b-datum.com/).


Instalação
===========

$ git clone https://github.com/b-datum/b-datum-backup-agent-linux-perl.git
$ cd b-datum-backup-agent-linux-perl
$ ./install.sh
$ source ~/perl5/perlbrew/etc/bashrc

Autor
=======

Gabriel Andrade <gabriel@b-datum.com>


Licensa e Copyright
===================

Licensa sob a GNU GPL v3.

Copyright (c) 2013, [b-datum](http://b-datum.com/)




Changes
b-datum (1.08) stable; urgency=medium

  * Support for progress bar
  * Read safely and consistent from cache 

 -- Thiago Rondon <thiago@b-datum.com>  Sat, 1 Nov 2014 23:44:00 -0300  

b-datum (1.07) stable; urgency=high

  * Fixed bdatum-restore dependency.

 -- Thiago Rondon <thiago@b-datum.com>  Thu, 10 Set 2014 13:15:00 -0300  

b-datum (1.06) stable; urgency=high

  * Fixed queuing, skipping file with no changes
  * New option, --use-stage and copy files to an area of stage, before the backup
  * --verbose and --debug is now just --debug
  * send a report with cause if we have problem to send a file

 -- Thiago Rondon <thiago@b-datum.com>  Thu, 7 Aug 2014 21:31:00 -0300

b-datum (1.05) stable; urgency=low

  * Support for multiples dirs in the path parameter per profile
  * Support for remote config for dirs.

 -- Gabriel Andrade <gabriel@b-datum.com>  Thu, 26 Jun 2014 02:56:00 -0300

b-datum (1.04) stable; urgency=low

  * Performance improvements: preforking workers in bdatum-backup.

 -- Thiago Rondon <thiago@b-datum.com>  Sat, 8 Mar 2014 01:11:10 -0300

b-datum (1.03) stable; urgency=low

  * retrocompatible config dir for config file in 
  * Use always .bdatum/backup.conf.tmp as temp file (this is for some *unix)
  * Shell-script for install and launch bdatum-backup in MacOS.
  * Use operation system in the version string.
  * Use proxy_port as option
  * Create ~/Library/LaunchAgents if necessary (MacOSX)
  * Allow proxy setup in backup.conf
  * Allow proxy without username and password
  * Add LWP::Protocol::connect for standard squid proxy to https
  * Select the date and time of the backup you want to restore.
  * Fixed parameter workers.

 -- Thiago Rondon <thiago@b-datum.com>  Wed, 22 Jan 2014 11:32:30 -0300

b-datum (1.02) stable; urgency=low

  * treat problems with DNS, be more user-friendly.

 -- Thiago Rondon <thiago@b-datum.com>  Mon, 27 Nov 2013 23:22:30 -0300

b-datum (1.01) stable; urgency=high

  * make check alive as the first job
  * 401 http unauthorized message in _send_request

 -- Thiago Rondon <thiago@b-datum.com>  Mon, 26 Nov 2013 13:24:10 -0300

b-datum (1.00) stable; urgency=low
  
  * Just renumbering

 -- Thiago Rondon <thiago@b-datum.com>  Mon, 25 Nov 2013 20:53:12 -0300

b-datum (0.26) unstable; urgency=low

  * removing dialog, using only bash.

 -- Thiago Rondon <thiago@b-datum.com>  Mon, 25 Nov 2013 19:00:32 -0300

b-datum (0.25) unsatable; urgency=high

  * Fix problems with deps for RPM distros.

 -- Thiago Rondon <thiago@b-datum.com>  Mon, 18 Nov 2013 10:12:48 -0300

b-datum (0.24) unstable; urgency=low
  
  * add LWP::MediaTypes and Module::Metadata in fatlib
  * fix kill USR just for parent_pid if its the case
  * Net::Amazon::S3 is not necessary for all cases
  * use $ENV{HOME}/.b-datum/backup.conf as alternative
  * check if system has net::amazon::s3
  * sending file size
  * adding config parameters for amazon.conf
  * s3safe -- multipart upload support
  * s3safe mode
  * cleanup b-datum-restore
  * make_control_file in Util.pm
  * Remove sys::syslog
  * add bdatum package for common constants
  * Module::Load is core just in 5.9, add to fatlib
  * use Module::Load::Conditional to know if system has sys::syslog (fedora)
  * add first tests and initial release of lib/bdatum/Util.pm
  * add debian/watch
  * Remove install.sh
  * src dir for sources, and add header.pl
  * remove TODO, we will use the github issues
  * BuildArch: noarch for RPM
  * blockquote to shell
  * update README.md with rpm package
  * turn off FileDigests and PayloadIsXz in rpmbuild for CentOS 5.4
  * AutoReqProv: no, please, dont try to find deps. We use fatpack.
  * add dialog as dep to rpm
  * we need ExtUtils::MakeMaker 6.76 to make RPM Packages
  * NO_PERLLOCAL and NO_PACKLIST in dh-make-perl
  * MIN_PERL_VERSION for 5.10
  * add spec and script for make rpm package
  * Remove Net::SSL from bdatum-restore

 -- Thiago Rondon <thiago@b-datum.com>  Fri, 08 Nov 2013 16:44:06 -0300

b-datum (0.23) unstable; urgency=low
  
  * package bdatum-backup and bdatum-restore with fatpack.
  * Just Net::SSLeay and Net::SSL as dep.

 -- Thiago Rondon <thiago@b-datum.com>  Sat, 07 Sep 2013 11:28:45 -0300

b-datum (0.22) unstable; urgency=low

  * disable cache in bdatum-restore
  * _validate_key return error if empty string
  * Add fatlib directory with pure perl modules
  * Add copyrights for modules embedded
  * JSON:PP is pure perl for deps
  * sending mtime metadata

 -- Thiago Rondon <thiago@b-datum.com>  Tue, 03 Sep 2013 19:28:53 -0300

b-datum (0.21) unstable; urgency=low

  * First version of bdatum-restore
  * At daemon&persistent mode, killing process when 401.
  * Remove DateTime
  * pairwise, stripped from List::MoreUtils source, removing a XS module.

 -- Thiago Rondon <thiago@b-datum.com>  Thu, 29 Aug 2013 23:50:26 -0300

b-datum (0.20) unstable; urgency=low

  * New name for package, just b-datum-linux.
  * Using Getopt::Long instead of ::Descriptive, a lot of dependencies (and XS)

 -- Thiago Rondon <thiago@b-datum.com>  Sun, 25 Aug 2013 11:42:34 -0300

b-datum (0.19) unstable; urgency=low

  * postinst: Don't run bdatum-config if box already has /etc/bdatum/backup.conf

 -- Thiago Rondon <thiago@b-datum.com>  Wed, 21 Aug 2013 19:54:18 -0300

b-datum (0.18) unstable; urgency=high

  * chomp to fix compare in _get_changes to symlink
  * check_alive to make agent linux online
  * Remove lastbackup file
  * Fix persistent mode with daemon.

 -- Thiago Rondon <thiago@b-datum.com>  Wed, 21 Aug 2013 13:29:53 -0300

b-datum (0.17) unstable; urgency=high

  * Fix to remove dirs
  * Put all dirs in the cache file
  * Fix for backup empty dirs
  * Fast second-first backup algorithm
  * _write_default_config, chmod 0400 $TARGET_FILE
  * error_handler for File::Next to _log_error

 -- Thiago Rondon <thiago@b-datum.com>  Sun, 18 Aug 2013 19:58:50 -0300

b-datum (0.16) unstable; urgency=medium
	
  * support for syslog
  * daemon mode

Makefile.PL
use ExtUtils::MakeMaker 6.76;

my $dir_libs = ['lib'];
$dir_libs = [] if -f '/etc/redhat-release';

WriteMakefile(
  NAME       => 'bdatum',
  AUTHOR     => 'bdatum-staff',
  DISTNAME   => 'bdatum',
  VERSION    => '1.08', # dh-make-perl, we can't use VERSION_FROM.
  ABSTRACT   => 'Provides tools for backup and restore files with b-datum.com',
  LICENSE    => 'GPLv3',
  EXE_FILES  => [ 'bin/bdatum-backup', 'bin/bdatum-restore', 'bin/bdatum-config', 'linux/b-datum.init' ],
  META_MERGE => {
    repository => {
      type => 'git',
      url  => 'git@github.com:b-datum/b-datum-linux.git',
      web  => 'https://github.com/b-datum/b-datum-linux'
    }
  },
  MIN_PERL_VERSION => '5.010',
  PREREQ_PM        => {
    'Net::SSLeay'        => 0,
    'IO::Socket::SSL'    => 0,
  },
  NO_PACKLIST => 1,
  NO_PERLLOCAL => 1,
  PMLIBDIRS => $dir_libs,
);

#sub MY::postamble {
#    q{
#    install:
#        cp linux/start-script /etc/init.d/b-datum
#};
#}

